enhancements/agent-installer/host-based-mirror.md

```diff
@@ -0,0 +1,282 @@
+
+### Goals
+
+- Remove the need to set up and maintain a local registry on a dedicated server
+  in fully-disconnected environments.
+- Include in the ISO the container images and tools required for mirroring.
+- Provide an additional command to build a container image tarfile using
+  the embedded release image in a connected environment.
+- When building the ISO in a disconnected environment, allow for a tarfile
+  built using `oc-mirror`, to be included.
+- Work with all topologies supported by the agent based installer - SNO,
+  Compact, HA.
+
+### Non-Goals
+
+- Replace the current method of using a local registry on a separate server
+  in a disconnected environment, as that method will still be supported.
+- Replace the current method in a connected environment where no local
+  registry is required, that will still be supported.
+- Reduce install times (since network access is greatly reduced it is expected
+  to reduce install times, but it is not a goal of this feature).
+- Fully support upgrades and updates, as this host-based mirror is intended
+  for Day 1 only.
+
+## Proposal
+
+### Workflow Description
+
+This change affects multiple functional areas of the agent installer - the
+user interface, image (ISO) creation, and host boot. Its not feasible to
+stage these modifications over multiple Openshift releases as they are all
+required for the host registry implementation.
+
+Currently, to perform a disconnected installation, a user must add the
+ImageContentSources section to install-config.yaml with the upstream sources
+and corresponding mirror locations. With this proposal, this step is no longer
+required, instead a tarfile containing the image sources must be included in
+the assets directory when running `openshift-install agent create image`.
+A new parameter flag `--embed-release-image=true` will be added to the `create
+image` command. When run in a connected environment, it will pull the images
+using the release image embedded in the installer. Alternatively, the user
+can generate this tarfile outside of the tool using `oc-mirror` to write to a
+file.  In either case, the ISO can be generated in a disconnected environment
+when the tarfile is available as it does not require access to upstream sources.
+
+Once the tarfile is provided, one of the challenges of this feature is how to
+include it in the ISO. A tarfile containing all of the container images for a
+release is on the order of 16GB. A separate ISO may be created to contain these
+images. The work that was done to create a separate partition for the
+[telco-ran-tools](https://github.com/openshift-kni/telco-ran-tools/blob/main/docs/partitioning.md)
+may be leveraged. In addition to the image tarfile, any tools needed during the
+host boot process (i.e. registry, `oc-mirror` etc.) must be included in the
+ignition when the ISO is created. This will utilize the work done for the
+[agent-tui](https://github.com/openshift/installer/pull/6777)
```

Comment:
didnt tui had size limitation?

Reply:
There's an implicit limitation, given by the available amount of space for the tmpfs ephemeral storage when booting the iso, since the content of the agent cpio archive (and thus currently the agent-tui) is loaded in memory

Reply:
Added that text to the document.